pkg/reconciler/delivery/policy.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delivery
+
+import (
+	"sort"
+	"fmt"
+)
+
+
+// Policy represents the rollout strategy used to update Route objects
+type Policy struct {
+	// Mode specifies the metric that the policy is based on
+	// Possible values are: "time", "request", "error"
+	Mode string
+
+	// Percents specifies the traffic percentages that the NEW Revision is expected to have
+	// at successive rollout stages; the list of integers must start at 0
+	// all entries must be in the range [0, 100), and must be sorted in increasing order
+	// Technically the final rollout percentage is 100, but this is implicitly understood,
+	// and should NOT be explicitly specified in Percents
+	// In addition to the traffic percentages, each stage can OPTIONALLY specify its own threshold
+	// this gives greater flexibility to policy design
+	// The threshold value for stage N is the value that must be achieved BEFORE moving to stage N+1
+	Percents []stage
```

Comment:
you have a public member with private type. That's not good. Is there a reason why is the type and fields public?

Reply:
Good question, I thought that maybe the Policy type and its members will have to be exposed outside the delivery package in the future, especially when we are going to implement the policy CRD, i.e. translating a user's yaml file into an actual `Policy` struct. Having said that, I realized I should also make `stage` public because otherwise outside packages will not be able to set the Percents field.